### PR TITLE
Timelock now shortens as user credit grows

### PR DESCRIPTION
**Pull request by asselstine**
- When timelock is used the user's credit is burned

### CODE REVIEW

File: contracts/prize-strategy/PrizeStrategy.sol [hunk 1]
```diff
@@ -90,19 +90,25 @@ contract PrizeStrategy is PrizeStrategyStorage,
...
+    uint256 credit = calculateAccruedCredit(user, balance);
     creditBalances[user] = CreditBalance({
       credit: uint256(creditBalances[user].credit).add(uint256(credit)).toUint128(),
       interestIndex: uint128(prizePool.interestIndexMantissa())
     });
   }
 
-  function calculateNewTicketCredit(address user, uint256 ticketBalance) internal returns (uint256) {
+  /// @notice Calculates the accrued interest for a user
+  function calculateAccruedCredit(address user, uint256 ticketBalance) internal returns (uint256) {
```
Comment (robsecord) [inline contracts/prize-strategy/PrizeStrategy.sol:111]: Missing @param's in docs

File: contracts/prize-strategy/PrizeStrategy.sol [hunk 2]
```diff
@@ -127,49 +133,134 @@ contract PrizeStrategy is PrizeStrategyStorage,
...
+      feeCredit = credit;
+    }
+    return feeCredit;
+  }
+
+  /// @notice Calculates the withdrawal unlock timestamp.  This function also accrues their ticket credit.
+  /// @param user The user who wishes to withdraw
+  /// @param token The token they are withdrawing
+  /// @return The absolute timestamp after which they are allowed to withdraw
+  function calculateWithdrawalUnlockTimestamp(address user, uint256 amount, address token) external override returns (uint256) {
```
Comment (robsecord) [inline contracts/prize-strategy/PrizeStrategy.sol]: We could add the variable name to the return statement `returns (uint256 timelock)` and then potentially save 2 lines of code in the body, as well as providing named return params to the caller.

File: contracts/prize-strategy/PrizeStrategy.sol [hunk 3]
```diff
@@ -128,48 +134,128 @@ contract PrizeStrategy is PrizeStrategyStorage,
...
+        // calculate how long it would take to accrue
+        timestamp = _currentTime().add(
+          _calculateAccrualTime(amount, remainingFee, previousPrize, previousPrizeAverageTickets, prizePeriodSeconds)
+        );
+      }
+      return timestamp;
+    }
+  }
+
+  function calculateAccrualTime(
```
Comment (robsecord) [inline contracts/prize-strategy/PrizeStrategy.sol:194]: Missing docs

File: contracts/prize-strategy/PrizeStrategy.sol (hunk 1 above)
Comment (asselstine) [reply contracts/prize-strategy/PrizeStrategy.sol:111]: :pray:

**Pull request merged**